plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,462 @@
+    aliases: [ custom_qos_policy_uuid ]
+    type: str
+  description:
+    description:
+    - The description of the Custom QoS Policy.
+    - Providing an empty string will remove the O(description="") from the Custom QoS Policy.
+    type: str
+  dscp_mappings:
+    description:
+    - The Differentiated Services Code Point (DSCP) mappings of the Custom QoS Policy.
+    type: list
+    elements: dict
+    suboptions:
+      dscp_from:
+        description:
+        - The DSCP range starting value.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ from ]
+      dscp_to:
+        description:
+        - The DSCP range ending value.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ to ]
+      dscp_target:
+        description:
+        - The DSCP target value.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ target ]
+      target_cos:
+        description:
+        - The target CoS to be driven based on the range of input values of DSCP coming into the fabric.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+      qos_priority:
+        description:
+        - The desired QoS class level to be used.
+        type: str
+        choices: [ level1, level2, level3, level4, level5, level6, unspecified ]
+        aliases: [ priority, prio ]
+  cos_mappings:
+    description:
+    - The CoS mappings of the Custom QoS Policy.
+    type: list
+    elements: dict
+    suboptions:
+      dot1p_from:
+        description:
+        - The Dot1P range starting value.
```

Comment:
should we mention something about traffic type in the description because I think "value" reflects the numerical representation of cos. see reference: https://notes.networklessons.com/qos-cos-vs-dscp

Reply:
same would also apply to dscp

Reply:
I change it to traffic type for CoS and encoding point for DSCP based on documentation.